cmd/mqtt/main.go

```diff
@@ -135,6 +138,24 @@ func main() {
 	}
 	defer nps.Close()
 
+	if cfg.mqttTargetHealthCheck != "" {
+		for {
+			res, err := http.Head(cfg.mqttTargetHealthCheck)
```

Comment:
Should you handle the response body close? Not closing it is usually considered to be a resource leak and prevents internal reuse of the underlying TCP connection.

Reply:
This is `HEAD` method http request, not `GET`. And there is no response body with HEAD. 
All I care is status code, so that's why decided to use it.

Reply:
Since `http.Head()` is just the shortcut for calling `DefaultClient.Do()`, there should be no difference between this and any other HTTP request. Since the [official docs (second paragraph)](https://golang.org/pkg/net/http/) say that the client is responsible for closing the response body, I would simply add after the error check:
```go
if err := res.Body.Close(); err != nil {
logger.Warn(....)
}
```
I doubt that there will be any problems with the current implementation, but I'd also like to respect the original design and the official docs.

Reply:
I actually used http.Get so I can log body response. And closing body of course.